src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -445,14 +444,14 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
     def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
         assert self.TYPE == daemon_spec.daemon_type
         deps: List[str] = []
-        hostnames: List[str] = []
-        for dd in self.mgr.cache.get_daemons_by_service('mgr'):
-            assert dd.hostname is not None
-            addr = self.mgr.inventory.get_addr(dd.hostname)
-            hostnames.append(addr)
+        daemons = self.mgr.cache.get_daemons_by_service('loki')
+        loki_host = ''
+        if daemons:
+            assert daemons[0].hostname is not None
+            loki_host = daemons[0].ip or self._inventory_get_addr(daemons[0].hostname)
```

Comment:
On testing this PR, Got this error - `Failed to apply: 'PromtailService' object has no attribute '_inventory_get_addr'` on starting the Promtail Service

Reply:
Thanks for comment!!

Reply:
Works fine now. Thank @hookak